3-UsingAPIs/Week2/homework/ex1-programmerFun/index.js

```diff
@@ -17,29 +17,31 @@ Full description at: https://github.com/HackYourFuture/Homework/blob/main/3-Usin
    url with `.shx`. There is no server at the modified url, therefore this 
    should result in a network (DNS) error.
 ------------------------------------------------------------------------------*/
-function requestData(url) {
-  // TODO return a promise using `fetch()`
+async function requestData(url) {
+  const response = await fetch(url);
+  if (!response.ok) {
+    throw 'network errors';
+  } else {
+    return response.json();
+  }
 }
 
 function renderImage(data) {
-  // TODO render the image to the DOM
-  console.log(data);
+  document.body.appendChild(document.createElement('img')).src = data.img;
 }
 
 function renderError(error) {
-  // TODO render the error to the DOM
+  document.body.appendChild(document.createElement('h1')).textContent = error;
```

Comment:
Although this works, for clarity I would split this into separate lines.

```suggestion
  const h1 = document.createElement('h1');
  h1.textContent = error.message;
  document.body.appendChild(h1);
```